src/dune/lib.ml

```diff
@@ -1542,7 +1562,17 @@ module Compile = struct
       t.requires
     in
     let lib_deps_info =
-      let pps = Lib_info.pps t.info in
+      let pps =
+        let resolve =
+          match db with
+          | None -> assert false (* FIXME ??? *)
```

Comment:
The `db` argument is `None` if `allow_overlaps` is `true` further below. Not sure what this is supposed to be for, so I didn't handle this case yet.

Reply:
I think it's just an artifact of using `Lib_name.t` for comparison.